packages/backend/src/tests/propertyBased/arbitraries/arbitraryWhereValue.ts

```diff
@@ -0,0 +1,114 @@
+import { fc } from '@fast-check/vitest';
+import { AllTablesRowsMap } from '../getTableRowsByTableName';
+import {
+    Schema,
+    getColumnDef,
+    getColumnPgType,
+    getTableDef,
+} from '@synthql/queries';
+
+export function arbitraryWhereValue<DB>({
+    schema,
+    allTablesRowsMap,
+    tableName,
+    columnName,
+    validWhere,
+}: {
+    schema: Schema<DB>;
+    allTablesRowsMap: AllTablesRowsMap;
+    tableName: string;
+    columnName: string;
+    validWhere: boolean;
+}): fc.Arbitrary<unknown> {
+    const tableRows = allTablesRowsMap.get(tableName);
+
+    const columnValues = tableRows?.map((row) => {
+        const value = row[columnName];
+
+        return value;
+    });
+
+    if (columnValues && columnValues?.length > 0) {
+        const columnValuesFromSet = Array.from(new Set(columnValues));
+
+        if (validWhere) {
+            return fc.constantFrom(...columnValuesFromSet);
+        } else {
+            const tableDef = getTableDef(schema, tableName);
+
+            const columnDef = getColumnDef(tableDef, columnName);
+
+            const columnPgType = getColumnPgType(columnDef);
+
+            // Return an arbitrary value that matches two conditions:
+            // 1. It is of the type of the PG column gotten from the schema
+            // 2. It currently does not exist in the column values array
+
+            // The matching arms are presently incomplete
+            // We need to include more infomration in the schema
+            // to make them work better
+
+            if (columnPgType === 'pg_catalog.int2') {
```

Comment:
These should all be arbitraries living in their own files. 

arbitraryInt2, arbitraryInt4, etc.

Reply:
or see my comment below for another suggestion on how to clean this up a bit